xarray/core/options.py

```diff
@@ -41,6 +69,21 @@ def __init__(self, **kwargs):
             raise ValueError('argument names %r are not in the set of valid '
                              'options %r' % (invalid_options, set(OPTIONS)))
         self.old = OPTIONS.copy()
+
+        if 'io_engines' in kwargs:
+            if isinstance(kwargs['io_engines'], basestring):
+                kwargs['io_engines'] = [kwargs['io_engines']]
+            new_engines = list()
+            for e in kwargs['io_engines']:
+                if e in _AVAILABLE_IO_ENGINES:
+                    new_engines.append(e)
+                else:
+                    raise ValueError('I/O engine %s not installed' % e)
+            if not new_engines:
+                raise ValueError('No I/O engines')
```

Comment:
I can see use cases for disabling the default IO engine altogether (e.g., if you want to always require an explicit choice). So let's not disallow this.

Reply:
This is allowed: `xarray.set_options(io_engines='pydap')`; this should not be allowed: `xarray.set_options(io_engines='foo')` unless `foo` engine is added to xarray.